src/mpid/ch4/shm/posix/release_gather/nb_release_gather.c

```diff
@@ -149,11 +149,13 @@ int MPIDI_POSIX_nb_release_gather_comm_init(MPIR_Comm * comm_ptr,
                 mpi_errno =
                     MPIDI_SHM_topology_tree_init(comm_ptr, 0,
                                                  MPIR_CVAR_BCAST_INTRANODE_TREE_KVAL,
-                                                 MPIDI_POSIX_Bcast_tree_type,
+                                                 MPIDI_POSIX_mpi_release_gather_get_tree_type
+                                                 (MPIR_CVAR_BCAST_INTRANODE_TREE_TYPE),
                                                  &nb_release_gather_info_ptr->bcast_tree,
                                                  &topotree_fail[0],
                                                  MPIR_CVAR_REDUCE_INTRANODE_TREE_KVAL,
-                                                 MPIDI_POSIX_Reduce_tree_type,
+                                                 MPIDI_POSIX_mpi_release_gather_get_tree_type
+                                                 (MPIR_CVAR_REDUCE_INTRANODE_TREE_TYPE),
```

Comment:
Are we assuming the CVAR always stay constant throughout the run?

How about make them two static global variables instead?

Reply:
Oh, I see. Let me fix it.

Reply:
Hi Hui, when should I define and initialize the static global variables?

Reply:
How about when you first initialize a tree?

Reply:
Hi Hui, I have changed MPIDI_POSIX_Bcast_tree_type and MPIDI_POSIX_Reduce_tree_type into static global variables. Would you please take another look?